cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/query/IndexUpdaterTest.java

```diff
@@ -563,36 +633,42 @@ public void indexSingleArrayFieldWhenIndexingArraysInSubDoc() throws Exception {
         pets.put("species", species);
         bodyMap.put("pet", pets);
         rev.setBody(DocumentBodyFactory.create(bodyMap));
-        DocumentRevision saved;
+        final DocumentRevision saved;
         saved = ds.createDocumentFromRevision(rev);
 
-        assertThat(IndexUpdater.updateIndex("basic", fields, db, ds, im.getQueue()), is(true));
+        assertThat(IndexUpdater.updateIndex("basic", fields, ds, im.getDatabaseQueue()), is(true));
         assertThat(getIndexSequenceNumber("basic"), is(1l));
 
-        cursor = null;
-        try {
-            SQLDatabase db = TestUtils.getDatabaseConnectionToExistingDb(this.db);
-            cursor = db.rawQuery(sql, new String[]{});
-            assertThat(cursor.getCount(), is(2));
-            assertThat(cursor.getColumnCount(), is(4));
-            List<String> petList = new ArrayList<String>();
-            while (cursor.moveToNext()) {
-                assertThat(cursor.columnName(0), is("_id"));
-                assertThat(cursor.getString(0), is("id123"));
-                assertThat(cursor.columnName(1), is("_rev"));
-                assertThat(cursor.getString(1), is(saved.getRevision()));
-                assertThat(cursor.columnName(2), is("name"));
-                assertThat(cursor.getString(2), is("mike"));
-                assertThat(cursor.columnName(3), is("pet.species"));
-                assertThat(cursor.getString(3), is(notNullValue()));
-                petList.add(cursor.getString(3));
+        im.getDatabaseQueue().submit(new SQLQueueCallable<Void>() {
+            @Override
+            public Void call(SQLDatabase db) throws Exception {
+                Cursor cursor = null;
+                try {
+                    cursor = db.rawQuery(sql, new String[]{});
+                    assertThat(cursor.getCount(), is(2));
+                    assertThat(cursor.getColumnCount(), is(4));
+                    List<String> petList = new ArrayList<String>();
+                    while (cursor.moveToNext()) {
+                        assertThat(cursor.columnName(0), is("_id"));
+                        assertThat(cursor.getString(0), is("id123"));
+                        assertThat(cursor.columnName(1), is("_rev"));
+                        assertThat(cursor.getString(1), is(saved.getRevision()));
+                        assertThat(cursor.columnName(2), is("name"));
+                        assertThat(cursor.getString(2), is("mike"));
+                        assertThat(cursor.columnName(3), is("pet.species"));
+                        assertThat(cursor.getString(3), is(notNullValue()));
+                        petList.add(cursor.getString(3));
+                    }
+                    assertThat(petList, containsInAnyOrder("cat", "dog"));
+                }catch (SQLException e) {
+                    Assert.fail(String.format("SQLException occurred executing %s: %s", sql, e));
```

Comment:
Any reason you chose to keep this catch block, but not the others (i.e. in previous tests)? The `SQLException` being thrown would end up making the test error via the `ExecutionException` from the call to `get()` anyway, but I guess it is vaguely nicer to have the fail rather than the error.
Anyway I think we should keep it the same in the tests either catch and fail, or just allow to bubble out and error.